packages/legend-application-studio/src/stores/EditorGraphState.ts

```diff
@@ -127,28 +138,59 @@ export class EditorGraphState {
 
   constructor(editorStore: EditorStore) {
     makeAutoObservable(this, {
+      mostRecentFormModeCompilationGraphHash: observable,
+      isStaleTextWarnings: observable,
       editorStore: false,
       graphGenerationState: false,
+      warnings: true,
       getPackageableElementType: false,
-      hasCompilationError: computed,
       clearCompilationError: action,
+      setWarnings: action,
+      setIsStaleTextWarnings: action,
+      setMostRecentFormModeCompilationGraphHash: action,
+      hasCompilationError: computed,
+      currentGraphHash: computed,
     });
 
     this.editorStore = editorStore;
     this.graphGenerationState = new GraphGenerationState(this.editorStore);
+    this.warnings = undefined;
+    this.isStaleTextWarnings = false;
+  }
+
+  setWarnings(val: EngineWarning[] | undefined): void {
+    this.warnings = val;
+  }
+
+  setIsStaleTextWarnings(val: boolean): void {
+    this.isStaleTextWarnings = val;
+  }
+
+  setMostRecentFormModeCompilationGraphHash(
+    graphHash: string | undefined,
+  ): void {
+    this.mostRecentFormModeCompilationGraphHash = graphHash;
```

Comment:
don't need this setter, we can just assign the value directly in the body of the `compile...()` generators